open_gpt/serve/executors/base.py

```diff
@@ -65,3 +69,33 @@ def generate(self, docs: 'DocumentArray', parameters: Dict = {}, **kwargs):
             d.tags['generated_text'] = self.model.generate(
                 prompt, max_context_length=max_context_length, **parameters
             )
+
+    @requests(on='/generate_stream')
+    def generate_stream(self, docs: 'DocumentArray', parameters: Dict = {}, **kwargs):
+        for k, v in parameters.items():
+            if k in ['top_k', 'num_return_sequences']:
+                parameters[k] = int(v)
+            if k == 'max_new_tokens':
+                logger.info(
+                    f"executor got max_new_tokens: {v} from gateway but it will be ignored")
+                parameters.pop('max_new_tokens')
+
+        for d in docs:
+            prompt = d.tags.get('input_ids') or d.tags.get('prompt') or d.text
+            past_key_values = pickle.loads(d.blob) if len(d.blob) > 0 else None
+            if not prompt:
+                continue
+
+            generated_text = self.model.step_generate(
+                prompt,
+                max_new_tokens=1,
+                past_key_values=past_key_values,
+                **parameters,
+            )
+            resp = next(generated_text)
+
+            d.blob = pickle.dumps(resp['past_key_values'])
+            d.tags['generated_text'] = resp['generated_text']
+            d.tags['output_ids'] = resp['output_ids']
```

Comment:
please make sure `output_ids` is a simple list of int. I doubt it will work by the current implementation.